src/app/auth/auth.service.ts

```diff
@@ -87,7 +88,23 @@ export class AuthService {
     this.auth0.parseHash((error, authResult) => {
       if (authResult && authResult.accessToken && authResult.idToken) {
         this.setSession(authResult);
-        this.store.dispatch(new UsersAction.GetCurrentUser());
+        this.store.dispatch(new UsersAction.GetCurrentUserWithLastLogin());
+        if (environment.analytics) {
+          this.store
+            .pipe(
+              select(selectCurrentUser),
+              filter(user => !!user && !!user.lastLoggedIn),
+              first(),
+              timeout(10000)
```

Comment:
I think this is useless when you add it after `first()` operator. You should move it before it.